rmw_opensplice_cpp/src/qos.cpp

```diff
@@ -189,6 +201,35 @@ bool get_datawriter_qos(
   return set_entity_qos_from_profile(qos_profile, datawriter_qos);
 }
 
+bool get_topic_qos(
+  DDS::DomainParticipant & participant,
+  const rmw_qos_profile_t & qos_profile,
+  DDS::TopicQos & topic_qos)
+{
+  bool ret = true;
+
+  DDS::ReturnCode_t status;
+  status = participant.get_default_topic_qos(topic_qos);
+  if (nullptr != check_get_default_topic_qos(status)) {
+    RMW_SET_ERROR_MSG(check_get_default_topic_qos(status));
```

Comment:
Wouldn't it better to store the result of `check_get_default_topic_qos()` rather than call it twice? Imagine if it was not idempotent (it is now, but if that were changed in the future this code would be broken).

Reply:
I know you moved this from the other file, but still the comment applies, might as well fix it if we can.